### PR TITLE
disallow dimensions smaller or equal 0 in shapes

### PR DESCRIPTION
**Pull request by spebern**
suggestion for fix of Bug #126800

### CODE REVIEW

File: src/core/Rakudo/Internals.pm [hunk 1]
```diff
@@ -402,8 +403,12 @@ my class Rakudo::Internals {
             if nqp::istype($_, Whatever) {
                 X::NYI.new(feature => 'Jagged array shapes');
             }
+            my $dim = $_.Int;
+            if $dim <= 0 {
+                X::IllegalDimensionInShape.new(dim => $dim).throw;
```
Comment (vendethiel) [inline src/core/Rakudo/Internals.pm]: `dim => $dim` is usually written `:$dim`

**Comment by lizmat** [comment]
935059ffb5a239d958cbc3ef addressed this, and some other stuff

**Comment by AlexDaniel** [comment]
Thanks, @spebern! However, #126800 cannot be closed without tests. Perhaps you could write some?

If I'm not mistaken, https://github.com/perl6/roast/blob/master/S09-multidim/decl.t is the right file for such tests.

**Comment by spebern** [comment]
Ok thanks for linking the file! I will write them today or tomorrow.

**Comment by AlexDaniel** [comment]
Already done by @usev6 in https://github.com/perl6/roast/commit/a2617480f4, but feel free to add more.

**Comment by AlexDaniel** [comment]
It disallows zero-sized arrays, why? ([RT #131965](https://rt.perl.org/Ticket/Display.html?id=131965))

**Pull request merged**